pyiceberg/io/pyarrow.py

```diff
@@ -1251,10 +1253,17 @@ def _task_to_record_batches(
                     arrow_table = arrow_table.filter(pyarrow_filter)
                     if len(arrow_table) == 0:
                         continue
-                    batch = arrow_table.to_batches()[0]
-            yield _to_requested_schema(
-                projected_schema, file_project_schema, batch, downcast_ns_timestamp_to_us=True, use_large_types=use_large_types
-            )
+                    output_batches = arrow_table.to_batches()
+                else:
+                    output_batches = iter([batch])
```

Comment:
nit: redundant with `output_batches` already initialized to the same value

Reply:
It is unfortunately different because of line 1245:

```
                batch = batch.take(indices)
```

Reply:
ah, didnt see that. 
Do you mind adding a comment here to mention the possible `batch` mutation? I feel like this might be an easy footgun later

Reply:
Moved the variable assignment to L1246 for readability